File: BotsSDK/korebot/src/main/java/kore/botssdk/audiocodes/webrtcclient/services/CallForegroundService.java
package kore.botssdk.audiocodes.webrtcclient.services;

import android.annotation.SuppressLint;
import android.app.Notification;
import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.os.IBinder;

import kore.botssdk.application.BotApplication;
import kore.botssdk.audiocodes.webrtcclient.General.Log;
import kore.botssdk.audiocodes.webrtcclient.General.NotificationUtils;

public class CallForegroundService extends Service {
    private static final String TAG = "CallForeService";

    private static final String START_FOREGROUND = "START_FOREGROUND";
    private static final String STOP_FOREGROUND = "STOP_FOREGROUND";

    @Override
    public void onCreate() {
        Log.d(TAG, "onCreate");
        super.onCreate();
    }

    @SuppressLint("ForegroundServiceType")
    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        if (intent != null) {
            Log.d(TAG, "Received intent: " + intent.getAction());
            String action = intent.getAction();
            if (START_FOREGROUND.equals(action)) {
                Log.d(TAG, "Received Start Foreground Intent ");
                Notification notification = NotificationUtils.addServiceNotification();
                startForeground(NotificationUtils.NOTIFICATION_SERVICE_ID, notification);

            } else if (STOP_FOREGROUND.equals(action)) {
                Log.d(TAG, "Received Stop Foreground Intent");
                stopForeground(true);
                stopSelf();
            }
        }
        return START_STICKY;
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Log.d(TAG, "onDestroy");
    }

    @Override
    public IBinder onBind(Intent intent) {
        // Used only in case of bound services.
        return null;
    }

    public static void startService(){
        Log.d(TAG, "startService foreground ");
        Context context = BotApplication.getGlobalContext();
        Intent startIntent = new Intent(context, CallForegroundService.class);
        startIntent.setAction(CallForegroundService.START_FOREGROUND);
        context.startService(startIntent);
    }

    public static void stopService(){
        Log.d(TAG, "stopService foreground ");
        Context context = BotApplication.getGlobalContext();
        Intent stopIntent = new Intent(context, CallForegroundService.class);
        stopIntent.setAction(CallForegroundService.STOP_FOREGROUND);
        context.stopService(stopIntent);
        NotificationUtils.removeServiceNotification();
    }

}

File: BotSdk/ui/src/main/java/com/kore/ui/audiocodes/webrtcclient/services/CallForegroundService.kt
package com.kore.ui.audiocodes.webrtcclient.services

import android.annotation.SuppressLint
import android.app.Service
import android.content.Context
import android.content.Intent
import android.os.IBinder
import android.util.Log
import com.kore.ui.audiocodes.webrtcclient.general.NotificationUtils
import com.kore.ui.audiocodes.webrtcclient.general.NotificationUtils.addServiceNotification
import com.kore.ui.audiocodes.webrtcclient.general.NotificationUtils.removeServiceNotification

class CallForegroundService : Service() {
    override fun onCreate() {
        Log.d(TAG, "onCreate")
        super.onCreate()
    }

    @SuppressLint("ForegroundServiceType")
    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        Log.d(TAG, "Received intent: " + intent.action)
        val action = intent.action
        if (START_FOREGROUND == action) {
            Log.d(TAG, "Received Start Foreground Intent ")
            val notification = addServiceNotification(this)
            startForeground(NotificationUtils.NOTIFICATION_SERVICE_ID, notification)
        } else if (STOP_FOREGROUND == action) {
            Log.d(TAG, "Received Stop Foreground Intent")
            stopForeground(true)
            stopSelf()
        }
        return START_STICKY
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d(TAG, "onDestroy")
    }

    override fun onBind(intent: Intent): IBinder? {
        // Used only in case of bound services.
        return null
    }

    companion object {
        private const val TAG = "CallForeService"
        private const val START_FOREGROUND = "START_FOREGROUND"
        private const val STOP_FOREGROUND = "STOP_FOREGROUND"
        fun startService(context: Context) {
            Log.d(TAG, "startService foreground ")
            val startIntent = Intent(context, CallForegroundService::class.java)
            startIntent.action = START_FOREGROUND
            context.startService(startIntent)
        }

        fun stopService(context: Context) {
            Log.d(TAG, "stopService foreground ")
            val stopIntent = Intent(context, CallForegroundService::class.java)
            stopIntent.action = STOP_FOREGROUND
            context.stopService(stopIntent)
            removeServiceNotification(context)
        }
    }
}